### PR TITLE
fix: support qdrant 0.8

### PR DESCRIPTION
**Pull request by dongxiang123**
Goals:

- ...This PR is aimed to fix the support for qdrant from Docarray
- ...
- [ ] check and update documentation, if required. See [guide](https://github.com/jina-ai/jina/blob/master/CONTRIBUTING.md#documentation-guidelines)

### CODE REVIEW

File: docarray/array/storage/qdrant/backend.py [hunk 1]
```diff
@@ -133,11 +134,11 @@ def _initialize_qdrant_schema(self):
                 full_scan_threshold=self._config.full_scan_threshold,
                 m=self._config.m,
             )
+            vectors_config = VectorParams(size=self.n_dim, distance=Distance.COSINE)
```
Comment (JoanFM) [inline docarray/array/storage/qdrant/backend.py:137]: the distance should be a parameter

File: docarray/array/storage/qdrant/backend.py [hunk 2]
```diff
@@ -175,10 +176,11 @@ def _get_offset2ids_meta(self) -> List[str]:
         ).result.payload['offset2id']
 
     def _update_offset2ids_meta(self):
+        vectors_config = VectorParams(size=1, distance=Distance.COSINE)
```
Comment (JoanFM) [inline docarray/array/storage/qdrant/backend.py:179]: same here

File: tests/unit/array/docker-compose.yml [hunk 3]
```diff
@@ -10,7 +10,7 @@ services:
       AUTHENTICATION_ANONYMOUS_ACCESS_ENABLED: 'true'
       PERSISTENCE_DATA_PATH: '/var/lib/weaviate'
   qdrant:
-    image: qdrant/qdrant:v0.7.0
+    image: qdrant/qdrant:v0.10.0
```
Comment (JoanFM) [inline tests/unit/array/docker-compose.yml:13]: please update to 0.8, not 0.10. 0.10 has spme other problems that will be dealt with in different PR

**Review by JoanFM — CHANGES_REQUESTED** [review]

File: tests/unit/array/docker-compose.yml (hunk 3 above)
Comment (JohannesMessner) [inline tests/unit/array/docker-compose.yml:13]: Are we sure we want to go all they way up to v10? I vaguely remember that there was some issue with qdrant 10, have we resolved that? @JoanFM

**Review by JohannesMessner — COMMENTED** [review]
With this we should also bump our `qdrant-client` dependency to version 8, no? It doesn't look like this preserves compatability with v7, but maybe I wrong

File: docarray/array/storage/qdrant/backend.py (hunk 2 above)
Comment (alaeddine-13) [reply docarray/array/storage/qdrant/backend.py:179]: not the same, this is just a hack for offset2id

File: tests/unit/array/docker-compose.yml (hunk 3 above)
Comment (dongxiang123) [reply tests/unit/array/docker-compose.yml]: Yes, with this, we can bump our client to version 8.

**Comment by dongxiang123** [comment]
> check there is another docker compose to change

Yes, that is right.

File: docarray/array/storage/qdrant/backend.py (hunk 2 above)
Comment (alaeddine-13) [reply docarray/array/storage/qdrant/backend.py]: @dongxiang123  no need to specify n_dim for offset2id collection.
Please re-apply your old changes in `_update_offset2ids_meta` method:
`vectors_config = VectorParams(size=1, distance=Distance.COSINE)`

File: setup.py [hunk 4]
```diff
@@ -102,6 +102,7 @@
             'onnxruntime',
             'jupyterlab',
             'transformers>=4.16.2',
+            'qdrant-client~=0.8.0',
```
Comment (JoanFM) [inline setup.py:105]: no need to add it here

**Review by JoanFM — CHANGES_REQUESTED** [review]

**Review by JoanFM — APPROVED** [review]
